client-v2/src/actions/ArticleFragments.ts

```diff
@@ -118,14 +111,15 @@ const maybeInsertGroupArticleFragment = (
           collection yourself.`,
           // if the user accepts then remove the moved item (if there was one),
           // remove article fragments past the cap count and finally persist
-          [
-            ...(removeAction ? [removeAction] : []),
+          (removeAction ? [removeAction] : []).concat([
+            insertGroupArticleFragment(id, index, articleFragmentId),
+            maybeAddFrontPublicationDate(articleFragmentId),
             addPersistMetaToAction(capGroupSiblings, {
               id: articleFragmentId,
               persistTo,
               applyBeforeReducer: true
             })(id, collectionCap)
-          ],
+          ]),
           // otherwise just undo the insertion and don't persist as nothing
```

Comment:
A quick note to say we shouldn't need this anymore! Yesss 🎉

Reply:
Good eyes 👀